### DISCUSSION TITLE
Why external library includes require full qualified path?

### DISCUSSION THREAD

**Discussion by iqbal-h** [mpromonet/webrtc-streamer#588] [General]
Hi,
When I add some header that is not part of webRTC, an error occurs at make. 

For example, I want to add opencv in desktopcapturer.cpp. 
In the CMakeList.txt, I added find_package(OpenCV REQUIRED). It successfully adds the path in CMakeCache.txt:
```
//The directory containing a CMake configuration file for OpenCV.
OpenCV_DIR:PATH=/usr/local/lib/cmake/opencv4
```
When I run make, following error occurs:
```
src/desktopcapturer.cpp:15: fatal error: opencv2/opencv.hpp: No such file or directory
    4 | #include <opencv2/opencv.hpp>
```

However, when I replace the include in desktopcapturer.cpp with `#include "/usr/local/include/opencv4/opencv2/opencv.hpp"`, on make, it recognizes the include. However, it starts throwing error on the includes inside opencv.hpp:
```
/usr/local/include/opencv4/opencv2/opencv.hpp:48:10: fatal error: opencv2/opencv_modules.hpp: No such file or directory
   48 | #include "opencv2/opencv_modules.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
There is no end to fixing these fully qualified paths.
How to fix this?

**Comment by iqbal-h** [comment]
@mpromonet would really appeciate your thoughts?

**Comment by mpromonet** [comment]
Hi @iqbal-h 

You need to add include path of opencv, something like `include_directories( ${OpenCV_INCLUDE_DIRS} )`

Best Regards,
Michel.

**Reply by iqbal-h** [reply]
Thanks
